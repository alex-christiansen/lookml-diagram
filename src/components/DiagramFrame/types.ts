import { ILookmlModel, ILookmlModelExplore, IGitBranch } from "@looker/sdk/lib/sdk/4.0/models"
import { SelectionInfoPacket, VisibleViewLookup } from "../interfaces"
import { getActiveGitBranch, getAvailGitBranches, DiagramError, DetailedModel } from "../../utils/fetchers"
import { OVERRIDE_KEY_SUBTLE, DIAGRAM_HEADER_HEIGHT } from "../../utils/constants"
import { DiagrammedModel } from "../../utils/LookmlDiagrammer/"
import styled from "styled-components"
import {
  Aside,
  Section,
  SelectOptionProps,
  Icon,
  theme,
  Header,
} from "@looker/components"

export interface DiagramFrameProps {
  unfilteredModels: ILookmlModel[]
  pathModelName: string
  pathExploreName: string
  modelDetail: DetailedModel
  dimensions: DiagrammedModel[]
  modelError: DiagramError
  setModelError: (e: DiagramError)=>void
  hiddenToggle: boolean,
  setHiddenToggle: (t: boolean)=>void
  displayFieldType: string,
  setDisplayFieldType: (s: string)=>void,
  selectedBranch: string,
  setSelectedBranch: (b: string)=>void,
 }

export interface DiagramHeaderProps {
  currentExplore: ILookmlModelExplore,
  selectionInfo: SelectionInfoPacket,
  toggleExploreInfo: ()=>void,
}
