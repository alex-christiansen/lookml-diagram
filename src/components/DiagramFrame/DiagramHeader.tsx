/*

 MIT License

 Copyright (c) 2021 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React from 'react'
import {
  Heading,
  Space,
  ButtonTransparent,
  IconButton,
  Button,
  Spinner,
  theme
} from '@looker/components'
import { Explore } from '@looker/icons'
import { useQueryClient } from 'react-query'
import { Info } from '@styled-icons/material-outlined/Info'
import { Close } from '@styled-icons/material-outlined/Close'
import styled from 'styled-components'
import { OVERRIDE_KEY_SUBTLE, OVERRIDE_KEY } from '../../utils/constants'
import { DiagramHeaderProps } from './types'
import { DiagramHeaderWrapper } from './FrameHelpers'

export const DiagramHeader: React.FC<DiagramHeaderProps> = ({
  currentExplore,
  selectionInfo,
  toggleExploreInfo,
  showExplorer,
  toggleExplorer,
  loadingQueryData,
  queryFields,
  setQueryFields
}) => {
  const exploreInfoStyles =
    selectionInfo.lookmlElement === 'explore'
      ? { color: OVERRIDE_KEY, backgroundColor: OVERRIDE_KEY_SUBTLE }
      : {}

  const ExplorerButton = showExplorer ? Button : ButtonTransparent

  return (
    <DiagramHeaderWrapper
      py="xsmall"
      px="large"
      className={currentExplore ? 'has-explore' : 'no-explore'}
    >
      <Space between>
        <Space gap="xsmall">
          <Heading as="h1" px="1rg">
            {currentExplore && currentExplore.label}
          </Heading>
          <IconButton
            label="Explore Info"
            icon={<Info />}
            onClick={toggleExploreInfo}
            style={exploreInfoStyles}
            size="large"
          />
        </Space>
        <Space gap="xsmall" justifyContent="flex-end">
          <ExplorerButton
            iconBefore={
              loadingQueryData ? (
                <Spinner
                  mr="xsmall"
                  color={showExplorer ? 'white' : OVERRIDE_KEY}
                  size={25}
                />
              ) : (
                <Explore />
              )
            }
            onClick={toggleExplorer}
          >
            Explore
          </ExplorerButton>
          {Object.keys(queryFields).length > 0 && (
            <IconButton
              label="Clear fields"
              icon={<Close />}
              onClick={() => setQueryFields({})}
            />
          )}
        </Space>
      </Space>
    </DiagramHeaderWrapper>
  )
}
