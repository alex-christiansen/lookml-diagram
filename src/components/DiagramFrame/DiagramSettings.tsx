/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React, { useContext } from "react"
import {
  SpaceVertical,
  Heading,
  FieldSelect,
  Label,
  Spinner,
} from "@looker/components"
import { X_INIT, Y_INIT, ZOOM_INIT } from '../../utils/constants/'
import { internalModelURL, internalExploreURL } from "../../utils/routes"
import { ILookmlModel, ILookmlModelExplore, IGitBranch } from "@looker/sdk/lib/sdk/4.0/models"
import { useHistory } from "react-router"
import { ExtensionContext } from "@looker/extension-sdk-react"
import { ExploreDropdown, DiagramSettingsProps } from "./types"
import {ExploreList, ExploreListitem, ExploreButton, SettingsPanel} from "./components"
import { getExploreListItemBackgroundColor } from "./utils"

export const DiagramSettings: React.FC<DiagramSettingsProps> = ({ 
  modelDetails,
  currentModel,
  modelName,
  setModelError,
  selectedBranch,
  setSelectedBranch,
  branchOpts,
  gitBranch,
  gitBranches,
  selectionInfo,
  exploreList,
  currentExplore,
  diagramExplore,
  setSelectionInfo,
  setViewVisible,
  setZoomFactor,
  setViewPosition,
  setMinimapUntoggled,
  setMinimapEnabled,
 }) => {
  const history = useHistory()
  return (
    <SettingsPanel width="275px" px="medium" py="large">
      <SpaceVertical>
        <Heading fontSize="large">Diagram Settings</Heading>
        <FieldSelect
          options={modelDetails}
          label="Choose a Model"
          placeholder="Select a model"
          value={modelName}
          onChange={(selectedModel: string) => {
            setModelError(undefined)
            setSelectedBranch("")
            history.push(internalModelURL({ model: selectedModel }))
          }
          }
          listLayout={{ maxHeight: 300 }}
          isLoading={modelDetails.length === 0 ? true : false}
        />
        {modelName && (
          <div>
            <SpaceVertical size="xxsmall">
              <FieldSelect
                options={branchOpts ? branchOpts : []}
                placeholder="Loading Git Branches..."
                label="Current Branch"
                value={gitBranch && gitBranch.name}
                onChange={(value)=>{setSelectedBranch(value)}}
                disabled={(gitBranch && gitBranch.is_production) || !diagramExplore}
              />
              <Label>Select an Explore</Label>
              <ExploreList>
                {exploreList && exploreList.map((explore: ExploreDropdown, index: number) => {
                  return (
                    <ExploreListitem key={`explore-${index}`} style={{backgroundColor: getExploreListItemBackgroundColor(explore.value, currentExplore, diagramExplore)}}>
                      <ExploreButton
                        onClick={() => {
                          selectionInfo.lookmlElement === "explore" || setSelectionInfo({})
                          setViewVisible({})
                          setZoomFactor(ZOOM_INIT)
                          setViewPosition({x: X_INIT, y: Y_INIT})
                          setMinimapUntoggled(true)
                          setMinimapEnabled(false)
                          history.push(
                            internalExploreURL({
                              model: currentModel.name,
                              explore: explore.value
                            })
                          )
                        }}
                        value={explore.value}
                      >
                        {explore.label}
                      </ExploreButton>
                    </ExploreListitem>
                  )
                })}
              </ExploreList>
            </SpaceVertical>
          </div>
        )}
      </SpaceVertical>
    </SettingsPanel>
  )
}
