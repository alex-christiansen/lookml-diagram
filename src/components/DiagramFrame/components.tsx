import { OVERRIDE_KEY_SUBTLE, DIAGRAM_HEADER_HEIGHT } from "../../utils/constants/"
import styled from "styled-components"
import {
  Aside,
  Section,
  Icon,
  Header,
} from "@looker/components"

export const Italics = styled.span`
font-style: italic;
`

export const SettingsPanel = styled(Aside as any)`
border-right: solid 1px ${(props) => props.theme.colors.ui2};
overflow-y: auto;
`

export const Stage = styled(Section as any)`
background: ${(props) => props.theme.colors.ui1};
overflow: hidden;
position: relative;
`
export const Rail = styled(Aside as any)`
border-right: solid 1px ${(props) => props.theme.colors.ui2};
align-items: center;
`

export const ExploreList = styled.ul`
margin: 0;
`

export const ExploreListitem = styled.li`
border-bottom: solid 1px ${(props) => props.theme.colors.ui2};
`

export const ExploreButton = styled.button`
all: inherit;
font-size: ${(props) => props.theme.fontSizes.small};
cursor: pointer;
padding: 12px 12px;
width: 100%;
border: none;


${Icon} {
  transform: translateX(0px);
  transition: all 500ms ease-out;
} 

&:hover {
  background-color: ${OVERRIDE_KEY_SUBTLE};
  
  ${Icon} {
    transform: translateX(4px);
  }

}

& > * {
  pointer-events: none;
}
`

export const ViewList = styled.ul`
margin: 0;
`
export const ViewListItem = styled.li`
border-bottom: solid 1px ${(props) => props.theme.colors.ui2};
`

export const ViewButton = styled.button`
all: inherit;
font-size: ${(props) => props.theme.fontSizes.small};
cursor: pointer;
padding: 12px 12px;
width: 100%;
border: none;

:hover {
  background-color: ${OVERRIDE_KEY_SUBTLE};
}

& > * {
  pointer-events: none;
}
`
export const DiagramHeaderWrapper = styled(Header as any)`
background-color: ${(props) => props.theme.colors.background};
border-bottom: solid 1px ${(props) => props.theme.colors.ui2};
transition: transform 500ms ease-out;
position: relative;
z-index: 1;

&.no-explore {
  transform: translateY(-${DIAGRAM_HEADER_HEIGHT}px);
}

&.has-explore {
  transform: translateY(0);
}
`
