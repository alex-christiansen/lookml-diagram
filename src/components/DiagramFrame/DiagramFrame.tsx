/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React, { useCallback } from "react"
import {
  SpaceVertical,
  IconButton,
  Layout,
} from "@looker/components"
import { SelectionInfoPacket, VisibleViewLookup } from "../interfaces"
import { useCurrentModel } from "../../utils/routes"
import { DiagrammedModel, DiagramMetadata } from "../../utils/LookmlDiagrammer/"
import "./styles.css"
import {MetadataPanel} from "./MetadataPanel/MetadataPanel"
import { ViewOptions, DiagramSettings, HelpPanel, ExploreDropdown } from "./FramePanels"
import { DiagramHeader } from "./DiagramHeader"
import { DiagramCanvas } from "./DiagramCanvas/DiagramCanvas"
import { 
  ZOOM_INIT,
  X_INIT,
  Y_INIT,
  OVERRIDE_KEY,
  OVERRIDE_KEY_SUBTLE
} from '../../utils/constants'
import { ILookmlModelExplore } from "@looker/sdk/lib/sdk/4.0/models"
import {DiagramFrameProps} from "./types"
import {Rail, Stage} from "./FrameHelpers"
import { prepareModelDropdown } from "./utils"

export const DiagramFrame: React.FC<DiagramFrameProps> = ({
  unfilteredModels,
  pathModelName,
  pathExploreName,
  modelDetail,
  dimensions,
  modelError,
  setModelError,
  hiddenToggle,
  setHiddenToggle,
  displayFieldType,
  setDisplayFieldType,
  selectedBranch,
  setSelectedBranch,
  }) => {
  const [viewVisible, setViewVisible] = React.useState<VisibleViewLookup>({})
  const [showSettings, setShowSettings] = React.useState(true)
  const [showHelp, setShowHelp] = React.useState(false)
  const [showViewOptions, setShowViewOptions] = React.useState(false)
  const [showNoAside, setShowNoAside] = React.useState(false)
  const [reload, setReload] = React.useState(false)
  const [selectionInfo, setSelectionInfo] = React.useState<SelectionInfoPacket>({})
  const [zoomFactor, setZoomFactor] = React.useState(ZOOM_INIT)
  const [viewPosition, setViewPosition] = React.useState({x: X_INIT, y: Y_INIT})
  const [minimapEnabled, setMinimapEnabled] = React.useState(false)
  const [minimapUntoggled, setMinimapUntoggled] = React.useState(true)

  const handleHiddenToggle = useCallback((event: any) => setHiddenToggle(event.target.checked), [])

  function showDiagramFullPage() {
    setShowHelp(false)
    setShowViewOptions(false)
    setShowSettings(false)
    setShowNoAside(true)
  }

  function isDiagramFullPage() {
    showDiagramFullPage()
    return true
  }

  function toggleExploreInfo() {
    if (currentExplore && modelDetail) {
      let selectInfo = selectionInfo.lookmlElement !== "explore" ? {lookmlElement: "explore"} : {}
      setSelectionInfo(selectInfo)
    }
  }

  const onlyDiagramSelected = () => { return showNoAside && !showViewOptions && !showSettings && !showHelp ? true : undefined }

  const modelDetails = prepareModelDropdown(unfilteredModels)

  let currentExplore: ILookmlModelExplore = modelDetail?.explores.filter((d: ILookmlModelExplore)=>{
    return d.name === pathExploreName
  })[0]

  const currentDimensions: DiagrammedModel = dimensions?.filter((d: DiagrammedModel)=>{
    return d.exploreName === pathExploreName
  })[0]

  const currentDiagramMetadata: DiagramMetadata = currentDimensions?.diagramDict

  let defaultViews: VisibleViewLookup = {}
  Object.keys(viewVisible).length === 0 && currentExplore && currentDiagramMetadata && Object.keys(currentDiagramMetadata.tableData)
  .map((lookmlViewName: string)=>{
    defaultViews[lookmlViewName] = true
  })
  Object.keys(viewVisible).length === 0 && currentExplore && currentDiagramMetadata && setViewVisible(defaultViews)

  return (
    <Layout hasAside height="100%">
      <Rail width="50px">
        <SpaceVertical style={{alignItems: "center"}} alignItems="center" gap="xsmall">
          <IconButton
            icon="Dashboard"
            label="Diagram"
            tooltipPlacement="right"
            size="large"
            toggle={onlyDiagramSelected()}
            onClick={showDiagramFullPage}
            style={{color: onlyDiagramSelected() && OVERRIDE_KEY, 
              backgroundColor: onlyDiagramSelected() && OVERRIDE_KEY_SUBTLE,
              borderRadius: "10px"}}
          />
          <IconButton
            icon="GearOutline"
            label="Settings"
            tooltipPlacement="right"
            size="large"
            onClick={() => isDiagramFullPage() && setShowSettings(!showSettings)}
            toggle={showSettings}
            style={{color: showSettings && OVERRIDE_KEY, 
              backgroundColor: showSettings && OVERRIDE_KEY_SUBTLE,
              borderRadius: "10px"}}
          />
          <IconButton
            icon="Tune"
            label="View Options"
            tooltipPlacement="right"
            size="large"
            onClick={() => isDiagramFullPage() && setShowViewOptions(!showViewOptions)}
            toggle={showViewOptions}
            style={{color: showViewOptions && OVERRIDE_KEY, 
              backgroundColor: showViewOptions && OVERRIDE_KEY_SUBTLE,
              borderRadius: "10px"}}
          />
          <IconButton
            icon="Help"
            label="Diagram Help"
            tooltipPlacement="right"
            id="diagram-help-btn"
            size="large"
            onClick={() => isDiagramFullPage() && setShowHelp(!showHelp)}
            toggle={showHelp}
            style={{color: showHelp && OVERRIDE_KEY, 
              backgroundColor: showHelp && OVERRIDE_KEY_SUBTLE,
              borderRadius: "10px", position: "absolute", bottom: "0px"}}
          />
        </SpaceVertical>
      </Rail>
      {showSettings && (
        <DiagramSettings
          modelDetails={modelDetails}
          modelName={pathModelName}
          currentModel={modelDetail}
          setModelError={setModelError}
          setSelectedBranch={setSelectedBranch}
          selectionInfo={selectionInfo}
          currentExplore={currentExplore}
          diagramExplore={pathExploreName}
          setSelectionInfo={setSelectionInfo}
          setViewVisible={setViewVisible}
          setZoomFactor={setZoomFactor}
          setViewPosition={setViewPosition}
          setMinimapUntoggled={setMinimapUntoggled}
          setMinimapEnabled={setMinimapEnabled}
        />
      )}
      {showViewOptions && (
        <ViewOptions
          displayFieldType={displayFieldType}
          hiddenToggle={hiddenToggle}
          viewVisible={viewVisible}
          setViewVisible={setViewVisible}
          handleHiddenToggle={handleHiddenToggle}
          setDisplayFieldType={setDisplayFieldType}
        />
      )}
      {showHelp && (
        <HelpPanel
        />
      )}
      <Stage>
        <DiagramHeader
          currentExplore={currentExplore}
          selectionInfo={selectionInfo}
          toggleExploreInfo={toggleExploreInfo}
        />
        <Layout hasAside height="100%" id="DiagramStage">
        <DiagramCanvas
          unfilteredModels={unfilteredModels}
          modelError={modelError}
          pathModelName={pathModelName}
          pathExploreName={pathExploreName}
          currentDimensions={currentDimensions}
          zoomFactor={zoomFactor}
          reload={reload}
          minimapUntoggled={minimapUntoggled}
          minimapEnabled={minimapEnabled}
          setZoomFactor={setZoomFactor}
          setViewPosition={setViewPosition}
          setReload={setReload}
          setMinimapUntoggled={setMinimapUntoggled}
          setMinimapEnabled={setMinimapEnabled}
          dimensions={currentDiagramMetadata}
          explore={currentExplore}
          selectionInfo={selectionInfo}
          setSelectionInfo={setSelectionInfo}
          hiddenToggle={hiddenToggle}
          displayFieldType={displayFieldType}
          viewVisible={viewVisible}
          viewPosition={viewPosition}
          selectedBranch={selectedBranch}
        />
        {currentExplore && Object.keys(selectionInfo).length > 0 && (
          <MetadataPanel
            explore={currentExplore}
            selectionInfo={selectionInfo}
            model={modelDetail.model}
          />
        )}
        </Layout>
      </Stage>
      </Layout>
  )
}
