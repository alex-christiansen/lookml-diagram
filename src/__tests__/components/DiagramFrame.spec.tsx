/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React from 'react';
import {
  mockModels,
  mockActiveGit,
  mockAvailGit,
  mockCurrentModel,
  mockSelectExplore,
  mockPathNames,
  mockMinimapDimensions } from "../test_data";
import 'jest-styled-components'
import { DiagramFrame } from '../../components/DiagramFrame/DiagramFrame'
import { shallow } from 'enzyme';

jest.mock('../../utils/fetchers', () => {
  return {
    useAllModels: jest.fn(() => {
      return mockModels
    }),
    getActiveGitBranch: jest.fn(() => {
      return mockActiveGit
    }),
    getAvailGitBranches: jest.fn(() => {
      return mockAvailGit
    }),
  }
})

jest.mock('../../utils/routes', () => {
  return {
    useCurrentModel: jest.fn(() => {
      return mockCurrentModel
    }),
    useSelectExplore: jest.fn(() => {
      return mockSelectExplore
    }),
    usePathNames: jest.fn(() => {
      return mockPathNames
    }),
  }
})

// jest.mock('../../utils/diagrammer', () => {
//   return {
//     getMinimapDimensions: jest.fn(() => {
//       return mockMinimapDimensions
//     }),
//   }
// })

// jest.mock("../../components/DiagramFrame/MetadataPanel/MetadataPanel", () => ({
//   MetadataPanel: () => "MetadataPanel"
// }))

// jest.mock("../../components/DiagramFrame/DiagramSettings", () => ({
//   DiagramSettings: () => "DiagramSettings"
// }))

// jest.mock("../../components/DiagramFrame/HelpPanel", () => ({
//   HelpPanel: () => "HelpPanel"
// }))

// jest.mock("../../components/DiagramFrame/ViewOptions", () => ({
//   ViewOptions: () => "ViewOptions"
// }))

// jest.mock("../../components/DiagramFrame/DiagramCanvas/Diagram", () => ({
//   Diagram: () => "Diagram"
// }))

// jest.mock("../../components/DiagramFrame/DiagramHeader", () => ({
//   DiagramHeader: () => "DiagramHeader"
// }))

// jest.mock("../../components/DiagramFrame/DiagramCanvas/DiagramToolbar", () => ({
//   DiagramToolbar: () => "DiagramToolbar"
// }))

// jest.mock("@looker/components", () => ({
//   Card: () => "Card",
//   Heading: () => "Heading",
//   Spinner: () => "Spinner",
//   SpaceVertical: () => "SpaceVertical",
//   IconButton: () => "IconButton",
//   Box: () => "Box",
//   Paragraph: () => "Paragraph",
//   Aside: () => "Aside",
//   Section: () => "Section",
//   Layout: () => "Layout",
//   Status: () => "Status",
//   theme: {colors: {key:"rgb(45, 126, 234)"}, space: {large: "2em"}, fontSizes: {large: "2em"}, fontWeights: {normal: "1em"}}
// }))

// describe('<DiagramFrame />', () => {
//   const default_loading = shallow(<DiagramFrame
//     unfilteredModels={mockModels}
//     pathModelName={"test"}
//     pathExploreName={"test_explore"}
//     modelDetail={mockModels[0]}
//     dimensions={[{exploreName: "text_explore", modelName: "test", diagramDict: view_dimensions}]}
//     hiddenToggle={true}
//     setHiddenToggle={undefined}
//     displayFieldType={"all"}
//     setDisplayFieldType={undefined} />);
//   it('default loading', () => {
//     expect(default_loading.debug()).toMatchSnapshot();
//   });

//   const show_help = shallow(<DiagramFrame
//     unfilteredModels={mockModels}
//     pathModelName={"test"}
//     pathExploreName={"test_explore"}
//     modelDetail={mockModels[0]}
//     dimensions={[{exploreName: "text_explore", modelName: "test", diagramDict: view_dimensions}]}
//     hiddenToggle={true}
//     setHiddenToggle={undefined}
//     displayFieldType={"all"}
//     setDisplayFieldType={undefined} />);
//   it('should help section', () => {
//     show_help
//     .find("IconButton#diagram-help-btn")
//     .simulate("click");
//     expect(show_help.debug()).toMatchSnapshot();
//   });

//   const show_settings = shallow(<DiagramFrame
//     unfilteredModels={mockModels}
//     pathModelName={"test"}
//     pathExploreName={"test_explore"}
//     modelDetail={mockModels[0]}
//     dimensions={[{exploreName: "text_explore", modelName: "test", diagramDict: view_dimensions}]}
//     hiddenToggle={true}
//     setHiddenToggle={undefined}
//     displayFieldType={"all"}
//     setDisplayFieldType={undefined} />);
//   it('should settings section', () => {
//     show_settings
//     .find("IconButton[label='Settings']")
//     .simulate("click");
//     expect(show_settings.debug()).toMatchSnapshot();
//   });
// });
