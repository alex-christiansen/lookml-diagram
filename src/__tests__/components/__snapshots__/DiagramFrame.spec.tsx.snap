// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiagramFrame /> default loading 1`] = `
"<Layout hasAside={true} height=\\"100%\\">
  <Styled(Aside) width=\\"50px\\" py=\\"xxsmall\\" pr=\\"xsmall\\">
    <SpaceVertical style={{...}} alignItems=\\"center\\" gap=\\"xsmall\\" ml=\\"xxsmall\\">
      <IconButton icon={{...}} label=\\"Settings\\" tooltipPlacement=\\"right\\" size=\\"large\\" onClick={[Function: toggleSettings]} toggle={true} style={{...}} />
      <IconButton icon={{...}} label=\\"View Options\\" tooltipPlacement=\\"right\\" size=\\"large\\" onClick={[Function: toggleViewOptions]} toggle={false} style={{...}} />
      <IconButton icon={{...}} label=\\"Diagram Help\\" tooltipPlacement=\\"right\\" id=\\"diagram-help-btn\\" size=\\"large\\" onClick={[Function: toggleHelp]} toggle={false} style={{...}} />
    </SpaceVertical>
  </Styled(Aside)>
  <DiagramSettings modelPathName=\\"test\\" explorePathName=\\"test_explore\\" modelDetails={{...}} exploreList={[undefined]} modelDetail={{...}} selectionInfo={{...}} currentExplore={[undefined]} diagramExplore={[undefined]} setSelectionInfo={[Function: bound _dispatchAction]} setViewVisible={[Function: bound _dispatchAction]} setZoomFactor={[Function: bound _dispatchAction]} setViewPosition={[Function: bound _dispatchAction]} setMinimapUntoggled={[Function: bound _dispatchAction]} setMinimapEnabled={[Function: bound _dispatchAction]} />
  <Styled(Section)>
    <DiagramHeader currentExplore={[undefined]} selectionInfo={{...}} toggleExploreInfo={[Function: toggleExploreInfo]} />
    <Layout hasAside={true} height=\\"100%\\" id=\\"DiagramStage\\">
      <DiagramCanvas modelDetail={{...}} unfilteredModels={{...}} pathModelName=\\"test\\" pathExploreName=\\"test_explore\\" currentDimensions={[undefined]} zoomFactor={0.5} reload={false} minimapUntoggled={true} minimapEnabled={false} setZoomFactor={[Function: bound _dispatchAction]} setViewPosition={[Function: bound _dispatchAction]} setReload={[Function: bound _dispatchAction]} setMinimapUntoggled={[Function: bound _dispatchAction]} setMinimapEnabled={[Function: bound _dispatchAction]} explore={[undefined]} selectionInfo={{...}} setSelectionInfo={[Function: bound _dispatchAction]} hiddenToggle={true} displayFieldType=\\"all\\" viewVisible={{...}} viewPosition={{...}} />
    </Layout>
  </Styled(Section)>
</Layout>"
`;

exports[`<DiagramFrame /> should help section 1`] = `
"<Layout hasAside={true} height=\\"100%\\">
  <Styled(Aside) width=\\"50px\\" py=\\"xxsmall\\" pr=\\"xsmall\\">
    <SpaceVertical style={{...}} alignItems=\\"center\\" gap=\\"xsmall\\" ml=\\"xxsmall\\">
      <IconButton icon={{...}} label=\\"Settings\\" tooltipPlacement=\\"right\\" size=\\"large\\" onClick={[Function: toggleSettings]} toggle={false} style={{...}} />
      <IconButton icon={{...}} label=\\"View Options\\" tooltipPlacement=\\"right\\" size=\\"large\\" onClick={[Function: toggleViewOptions]} toggle={false} style={{...}} />
      <IconButton icon={{...}} label=\\"Diagram Help\\" tooltipPlacement=\\"right\\" id=\\"diagram-help-btn\\" size=\\"large\\" onClick={[Function: toggleHelp]} toggle={true} style={{...}} />
    </SpaceVertical>
  </Styled(Aside)>
  <HelpPanel />
  <Styled(Section)>
    <DiagramHeader currentExplore={[undefined]} selectionInfo={{...}} toggleExploreInfo={[Function: toggleExploreInfo]} />
    <Layout hasAside={true} height=\\"100%\\" id=\\"DiagramStage\\">
      <DiagramCanvas modelDetail={{...}} unfilteredModels={{...}} pathModelName=\\"test\\" pathExploreName=\\"test_explore\\" currentDimensions={[undefined]} zoomFactor={0.5} reload={false} minimapUntoggled={true} minimapEnabled={false} setZoomFactor={[Function: bound _dispatchAction]} setViewPosition={[Function: bound _dispatchAction]} setReload={[Function: bound _dispatchAction]} setMinimapUntoggled={[Function: bound _dispatchAction]} setMinimapEnabled={[Function: bound _dispatchAction]} explore={[undefined]} selectionInfo={{...}} setSelectionInfo={[Function: bound _dispatchAction]} hiddenToggle={true} displayFieldType=\\"all\\" viewVisible={{...}} viewPosition={{...}} />
    </Layout>
  </Styled(Section)>
</Layout>"
`;

exports[`<DiagramFrame /> should settings section 1`] = `
"<Layout hasAside={true} height=\\"100%\\">
  <Styled(Aside) width=\\"50px\\" py=\\"xxsmall\\" pr=\\"xsmall\\">
    <SpaceVertical style={{...}} alignItems=\\"center\\" gap=\\"xsmall\\" ml=\\"xxsmall\\">
      <IconButton icon={{...}} label=\\"Settings\\" tooltipPlacement=\\"right\\" size=\\"large\\" onClick={[Function: toggleSettings]} toggle={false} style={{...}} />
      <IconButton icon={{...}} label=\\"View Options\\" tooltipPlacement=\\"right\\" size=\\"large\\" onClick={[Function: toggleViewOptions]} toggle={false} style={{...}} />
      <IconButton icon={{...}} label=\\"Diagram Help\\" tooltipPlacement=\\"right\\" id=\\"diagram-help-btn\\" size=\\"large\\" onClick={[Function: toggleHelp]} toggle={false} style={{...}} />
    </SpaceVertical>
  </Styled(Aside)>
  <Styled(Section)>
    <DiagramHeader currentExplore={[undefined]} selectionInfo={{...}} toggleExploreInfo={[Function: toggleExploreInfo]} />
    <Layout hasAside={true} height=\\"100%\\" id=\\"DiagramStage\\">
      <DiagramCanvas modelDetail={{...}} unfilteredModels={{...}} pathModelName=\\"test\\" pathExploreName=\\"test_explore\\" currentDimensions={[undefined]} zoomFactor={0.5} reload={false} minimapUntoggled={true} minimapEnabled={false} setZoomFactor={[Function: bound _dispatchAction]} setViewPosition={[Function: bound _dispatchAction]} setReload={[Function: bound _dispatchAction]} setMinimapUntoggled={[Function: bound _dispatchAction]} setMinimapEnabled={[Function: bound _dispatchAction]} explore={[undefined]} selectionInfo={{...}} setSelectionInfo={[Function: bound _dispatchAction]} hiddenToggle={true} displayFieldType=\\"all\\" viewVisible={{...}} viewPosition={{...}} />
    </Layout>
  </Styled(Section)>
</Layout>"
`;
